sound/soc/codecs/rt715.c

```diff
@@ -714,9 +714,16 @@ static const struct snd_soc_dapm_route rt715_audio_map[] = {
 static int rt715_set_bias_level(struct snd_soc_component *component,
 				enum snd_soc_bias_level level)
 {
+	struct rt715_priv *rt715 = snd_soc_component_get_drvdata(component);
 	struct snd_soc_dapm_context *dapm =
 		snd_soc_component_get_dapm(component);
 
+	if (!rt715->hw_init) {
+		dev_err(component->dev,
+			"%s called before hw_init completion\n", __func__);
+		return -EINVAL;
+	}
```

Comment:
This may not be necessary. The set_bias_level ops will not be called before the codec is probed. And rt715_probe will not be completed if rt715->hw_init is false.